File: 014_Longest_Common_Prefix/014_1.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";
        String first = strs[0];
        for (int i = 1; i < strs.length; ++i) {
            // while (strs[i].indexOf(first) != 0)
            while (!strs[i].startsWith(first)) {
                first = first.substring(0, first.length() - 1);
                if (first.isEmpty()) return "";
            }
        }
        return first;
    }
}

public class leetcode {
    public static void main(String[] args) {
        String[] strs = {"flower", "flow", "flight"};
        Solution s = new Solution();
        System.out.println(s.longestCommonPrefix(strs));
    }
}

File: 014_Longest_Common_Prefix/014_1.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs: 
            return ""
        s1 = min(strs)
        s2 = max(strs)
        for i, x in enumerate(s1):
            if x != s2[i]:
                return s1[:i]
        return s1


if __name__ == "__main__":
    strs = ["flow", "flower", "flight"]
    s = Solution()
    print(s.longestCommonPrefix(strs))